CMakeLists.txt

```diff
@@ -41,6 +41,30 @@ set(VERSION_MAJ_MIN "${VERSION_MAJOR}.${VERSION_MINOR}")
 #                               Compiler Setup
 #------------------------------------------------------------------------------
 
+# Compiler requirements
+set(CLANG_MINIMUM_VERSION 6.0)
+set(APPLE_CLANG_MINIMUM_VERSION 9.1)
+set(GCC_MINIMUM_VERSION 8.0)
+
+# Ensures that the specified C++ compiler meets our version requirements.
+macro(check_compiler_version compiler minimum_version)
+  try_run(program_result
```

Comment:
I am aware that this check can be placed outside the macro. But I don't think it adds a noticeable slowdown and improves code encapsulation.